src/filehash.rs

```diff
@@ -0,0 +1,90 @@
+//! Everything related to wikis filehash functionality
+
+use InputPaths;
+use error::*;
+use std::fs::{self, File};
+use std::path::Path;
+use std::io::{Write, Read, BufReader, BufRead};
+use uuid::{Uuid, NAMESPACE_DNS};
+
+pub struct Filehash;
+
+impl Filehash {
+    /// Reads the file hash for the file specified by `file_str` out of `hash_file_str`
+    fn read_file_hash(hash_file_str: &str, file_str: &str) -> Option<String> {
+        if let Ok(hash_file_res) = File::open(hash_file_str) {
+            let hash_file_reader = BufReader::new(hash_file_res);
+            for line in hash_file_reader.lines() {
+                if let Ok(l) = line {
+                    // Break the line between `<hash>:<file>`
+                    let sha_args: Vec<&str> = l.split(':').collect();
+
+                    // File matched
+                    if sha_args[1] == file_str {
```

Comment:
I would use [get()](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get) here since you can catch the out of bounds access to the vector.